src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -607,74 +612,55 @@ static int guigetstr(Object* ho, char** cpp) {
     return 1;
 }
 
-static PyObject* loads;
-static PyObject* dumps;
+static nb::callable loads;
+static nb::callable dumps;
 
 static void setpickle() {
     if (dumps) {
         return;
     }
-    PyObject* pickle = PyImport_ImportModule("pickle");
-    if (pickle) {
-        Py_INCREF(pickle);
-        dumps = PyObject_GetAttrString(pickle, "dumps");
-        loads = PyObject_GetAttrString(pickle, "loads");
-        if (dumps) {
-            Py_INCREF(dumps);
-            Py_INCREF(loads);
-        }
-    }
+    nb::module_ pickle = nb::module_::import_("pickle");
+    dumps = pickle.attr("dumps");
+    loads = pickle.attr("loads");
     if (!dumps || !loads) {
```

Comment:
I've seen `.is_valid()` being used in other places where we want to check that the C-API didn't return a `nullptr`. Do you know what the difference is?

Reply:
This is the same.

`is_valid()` is part of `api` class and does `derived().ptr() != nullptr`.

The `operator bool()` is part of `handle` class and does `m_ptr != nullptr`.